internal/appsec/waf.go

```diff
@@ -188,6 +212,34 @@ func newGRPCWAFEventListener(handle *waf.Handle, _ []string, timeout time.Durati
 			mu     sync.Mutex // events mutex
 		)
 
+		wafCtx := waf.NewContext(handle)
+		if wafCtx == nil {
+			// The WAF event listener got concurrently released
+			return
+		}
+
+		// The same address is used for gRPC and http when it comes to client ip
+		values := map[string]interface{}{}
+		for _, addr := range addresses {
+			if addr == httpClientIPAddr && handlerArgs.ClientIP.IsValid() {
+				values[httpClientIPAddr] = handlerArgs.ClientIP.String()
+			}
+		}
+
+		matches, actionIds := runWAF(wafCtx, values, timeout)
+		wafCtx.Close()
```

Comment:
can you please defer this call right after line 219? This avoids leaking memory in case any of the lines between 219 and this one panic in loop on each request.